libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -632,6 +632,33 @@ int AP_HAL__I2CDevice_read_registers(lua_State *L) {
     return success;
 }
 
+int AP_HAL__UARTDriver_readstring(lua_State *L) {
+    const int args = lua_gettop(L);
+    if (args != 2) {
+        return luaL_argerror(L, args, "expected 1 arguments");
+    }
+
+    AP_HAL::UARTDriver * ud = *check_AP_HAL__UARTDriver(L, 1);
+
+    const uint16_t count = get_uint16_t(L, 2);
+    uint8_t *data = (uint8_t*)malloc(count);
```

Comment:
You could use `luaM_malloc` here, that would allocate out of lua memory and remove the need for the nullptr check because it will error out.

Reply:
i think using malloc() is OK and needs less lua dedicated memory